test/XRP/common-xrpl-client.test.ts

```diff
@@ -214,15 +214,10 @@ describe('Common XRPL Client', function (): void {
       wallet,
     )
 
-    // TODO: (amiecorso) in separate PR, update the generation of a `TransactionResult` such that:
-    // - consider renaming to FinalTransactionResult
```

Comment:
i *think* i'm ok with this - however, does this mean that we think there is no value in having `TransactionResult` represent non-finalized (intermediate) outcomes?

another way we could do this while maintaining that flexibility is adding a flag to `TransactionResult` called `isComplete` or something

Reply:
I actually just realized this this morning that I think `TransactionResult` (now `FinalTransactionResult`) is actually used elsewhere in non-reliable-sub places... thus making it distinctly *not* final so lemme think about the best way to address that.

Reply:
Ugh yeah so the *one* issue with calling this `FinalTransactionResult` is that this type is returned from methods in the `DefaultXrpClient`, which doesn't implement any reliable sending... It may be an internal class, but that becomes unacceptably ambiguous.  I think I need to just change the name back.... adding a flag to transactionResult is probably the move.

Reply:
did all that